docs/reduction-in-locks.md

```diff
@@ -0,0 +1,98 @@
+
+The new backup design includes the following phases:
+
+### Phase 1: `Operations performed without the lock`
+
+1. Copy all redo logs from the checkpoint up to the current `LSN` and start tracking new entries.
+2. Start the redo log thread.
+3. Track file operations from the redo log by parsing the `MLOG_FILE_*` records.
+4. Copy of all the `.ibd` files.
+
+### Phase 2: `Operations performed under the lock`
+
+1. Acquire the backup lock (Lock Tables For Backup (LTFB)/Lock Instance For Backup (LIFB)) to prevent new DDL operations, such as creating or altering tables.
+2. Check the file operations that were tracked and recopy the tablespaces.
+3. Create additional `meta` files to perform the required actions (deletions or renames) on the already copied files. This approach ensures that the backup remains consistent and accurate without disrupting the streaming process. This step is required for taking streaming backups.
+    
+    The meta files are the following:
+
+    | File         | Description                                                                                                      |
+    |--------------|------------------------------------------------------------------------------------------------------------------|
+    | `.new`       | For files that need to be recopied due to encryption or `ADD INDEX` operations.                                  |
+    | `.del`       | For deleted files. If a file has been copied, create a `space_id.del` file.                                      |
+    | `.ren`       | For renamed files after copying. Create a `space_id.ren` file.                                                   |
+    | `.crpt`      | For files that cannot be fully copied due to encryption changes. These will be recopied, and a `.crpt` file will be created for the existing file. |
+    | `.new.meta`  | Similar to `.new`, but for incremental backups.                                                                  |
+    | `.new.delta` | Similar to `.new`, but for incremental backups. Create `t1.ibd.meta` and `t1.ibd.delta` instead of `t1.ibd`.      |
+
+4. Gather a sync point from all engines (InnoDB LSN, binlog, GTID, etc.) by querying the `log_status`.
+5. Stop the redo flow thread once it copies at least up to sync point at step 4.
+6. Release the backup lock (LTFB/LIFB).
+
+### Phase 3. Processing the new metadata files during the `--prepare` phase before crash recovery starts.
+
+1. The `.crpt` files are removed matching the name after stripping the extension. This step is crucial before the IBD scan because these files are incomplete (they could even be zero size).
```

Comment:
These steps need standardization - either start all with a verb (which is preferred) or do not. Do not mix.

Reply:
updated